src/fragments/lib/graphqlapi/js/working-with-files.mdx

```diff
@@ -0,0 +1,984 @@
+The Storage and GraphQL API categories can be used together to associate a file, such as an image or video, with a particular record. For example, you might create a `Person` model with a profile picture, or a `Post` model with an associated image. With Amplify's GraphQL API and Storage categories, you can reference the file within the model itself to create an association.
+
+To get started, run the following in an existing Amplify project:
+
+```bash
+npm i @aws-amplify/ui-react  #For Authenticator component
+amplify add auth             #Select default configuration
+amplify add storage          #Select S3 with read/write access
+amplify add api              #Select Cognito User Pool for authorization type
+```
+
+When prompted, use the following schema:
+
+```graphql
+type Song @model {
+  id: ID!
+  name: String!
+  covertArtKey: String
+}
+```
+
+Save the schema and run `amplify push` to deploy the changes.
+
+To use files and attachments you need authorization credentials for reading and writing to Amazon S3, which `amplify add auth` configures in the default setting along with a Cognito User Pool. For more details on Storage / API authorization, see the [Configuring Authorization](#configuring-authorization) section below.
+
+For the complete working example, including required imports and React component state management, see the [Complete Example](#complete-example) below.
+
+## Creating a record with a GraphQL API
+
+Running the following mutation will create a record with the GraphQL API. The corresponding file will be associated with this record.
+
+<BlockSwitcher>
```

Comment:
This example is maybe too minimal, as customer are able to get similar code snippets from other areas of our docs. I think we can expand the current section to "Create a record with an attached file". I.e. also show how to associate the file.

For the reader, we should answer the question: "What's Amplify's recommended guidance? Upload file first, then create record or vice versa?"

Reply:
To clarify - your suggestion is that we combine record creation, file upload, and file association with the record into a single function?

Reply:
Into a single "section". Re: if in the same function, I think we should just develop an opinion that customers should generally follow. If we think diff functions are better then let's do that.

Reply:
Re: order of operations, it depends on the customer use-case. If, for instance, there is a situation where a created record will always have an initial file, then we should perform the upload, then the mutation. In a situation where the file may be added later (for instance, adding a profile picture after a profile has been created), then we would create the record, upload the file, then update the record. I'll call this out in the documentation to make it clear.